src/main/java/Functionality.java

```diff
@@ -0,0 +1,93 @@
+    public Functionality() {
+        tasks = new ArrayList<>();
+    }
+
+    public void enter(String s) {
+        if (s.equals("list")) {
+            list();
+        } else if (s.length() > 5 && s.substring(0,4).equals("done")) {
+            done(Integer.parseInt(s.substring(5)));
+        } else if (s.length() > 6 && s.substring(0,6).equals("delete")) {
+            delete(Integer.parseInt(s.substring(7)));
+        } else if (s.length() < 5 || !s.contains(" ") || s.stripTrailing().length() < 9) {
+            error(s);
+        } else {
+            add(s);
+        }
+    }
+
+
+    public void list() {
+        System.out.println(linedivider + "Here are the tasks in your list:");
+        int entryno = 1;
+        for (Task ls : tasks) {
+            System.out.println(entryno + ". " + ls);
+            entryno++;
+        }
+        System.out.println(linedivider);
+    }
+
+    public void add(String s) {
+        int whitespaceidx = s.indexOf(" ");
+        String taskType = s.substring(0, whitespaceidx);
+        if (taskType.equals("todo")) {
+            String theTask = s.substring(whitespaceidx + 1);
+            Todo t = new Todo(theTask, tasks.size() + 1);
+            tasks.add(t);
+            System.out.println(linedivider + "Got it. I've added this task:\n " + t
+                    + "\nNow you have " + tasks.size() + " tasks in the list.\n" + linedivider);
+        } else {
+            int taskIdx = s.indexOf("/");
+            String theTask = s.substring(whitespaceidx + 1, taskIdx - 1);
+            String date = s.substring(taskIdx + 1);
+            if (taskType.equals("deadline")) {
+                Deadline t = new Deadline(theTask, date, tasks.size() + 1);
+                tasks.add(t);
+                System.out.println(linedivider + "Got it. I've added this task:\n " + t
+                        + "\nNow you have " + tasks.size() + " tasks in the list.\n" + linedivider);
+            } else if (taskType.equals("event")) {
+                Event t = new Event(theTask, date, tasks.size() + 1);
+                tasks.add(t);
+                System.out.println(linedivider + "Got it. I've added this task:\n " + t
+                        + "\nNow you have " + tasks.size() + " tasks in the list.\n" + linedivider);
+            }
+
+        }
+    }
+
+    public void done(int n) {
+        tasks.get(n-1).makeDone();
+        System.out.println(linedivider + "Nice! I've marked this task as done:\n" + tasks.get(n-1)
+                + "\n" + linedivider);
+    }
+
+    public void delete(int n) {
+        Task rm = tasks.remove(n - 1);
+        System.out.println(linedivider + "Noted. I've removed this task:\n" + rm + "\n"
+                + "Now you have " + tasks.size() + " tasks in the list.\n" + linedivider);
+    }
+
+    public void error(String s) {
+        System.out.println(linedivider);
+        if (s.length() >= 4 && (s.stripTrailing().equals("todo")
```

Comment:
Just something that I've thought, since your error code for description that cannot be empty is repeated, perhaps it might be good to just use it one time. eg. `System.out.println(":( OOPS!!! The description of " + command + " cannot be empty.");`, where command is todo/event/deadline